### PR TITLE
Reproduce #1085

### PR DESCRIPTION
**Pull request by forki**

### CODE REVIEW

**Comment by forki** [comment]
mhm so the jenkins build is read, but I can't find the expected error.
What am I doing wrong?

**Comment by forki** [comment]
Ok appveyor is green. This is even more weird. Shouldn't this one at least run compiler tests?

File: src/fsharp/FSharp.Compiler.Unittests/RecursiveRecords.fs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    Name: string
+    Type: TypeDef
+}
+
+let Int = { Name = "Int"; Fields = [] }
+let Complex = { Name = "Complex"; Fields = [{ Name = "X"; Type = Int}] }
+Complex.Fields <- { Name = "Deep"; Type = Complex } :: Complex.Fields
+
+[<Test>]
+let ``recursive structure with same record reference should not satck overflow`` () =
```
Comment (OkayX6) [inline src/fsharp/FSharp.Compiler.Unittests/RecursiveRecords.fs]: Typo, ``satck overflow``=> ``stack overflow`` :)

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

**Comment by KevinRansom** [comment]
@forki according to the appveyor log the tests did build including your change and run and pass.

Jenkins was in a bad way following the coreclr merge, and it only ever built things.

I have now fixed Jenkins to build and run the same test suites as Appveyor runs.  I'm not sure how reliable our tests are under Jenkins right now so we will keep appveyor and Jenkins plugged in, until we have a lot more confidence in Jenkins.

I am taking a look at your PR to see where the issue is.

**Comment by forki** [comment]
there might also be a slight chance that this case is already fixed and we
need to come up with an example that can't be solved by simple reference
equality test - cc @dsyme

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

**Comment by KevinRansom** [comment]
@dotnet-bot test this please

**Comment by KevinRansom** [comment]
Sorry for the spam :-)

**Pull request merged**